packages/@sanity/image-url/src/urlForImage.js

```diff
@@ -61,15 +61,18 @@ export default function urlForImage(options) {
 }
 
 // Convert an asset-id, asset or image to an image record suitable for processing
-function parseSource(source) {
+export function parseSource(source) {
   let image
 
   // Did we just get an asset id?
   if (typeof source == 'string') {
     image = {
-      asset: {_ref: image}
+      asset: {_ref: source}
     }
-  } else if (source._type == 'sanity.imageAsset') {
+  } else if (
+    source._type == 'sanity.imageAsset' ||
+    (typeof source == 'object' && typeof source._ref == 'string')
```

Comment:
Could you use triple equals (`===`) for these checks?

Reply:
Sure, I didn't want to change convention.

Reply:
We should probably fix the other occurrences as well, but leave that up to us. Seems like an oversight on our part